internal/connector/internal/api/dbapi/processor.go

```diff
@@ -0,0 +1,143 @@
+package dbapi
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/db"
+)
+
+type ProcessorDesiredState string
+type ProcessorStatusPhase string
+
+const (
+	ProcessorUnassigned ProcessorDesiredState = "unassigned"
+	ProcessorReady      ProcessorDesiredState = "ready"
+	ProcessorStopped    ProcessorDesiredState = "stopped"
+	ProcessorDeleted    ProcessorDesiredState = "deleted"
+
+	ProcessorStatusPhaseAssigning      ProcessorStatusPhase = "assigning"      // set by kas-fleet-manager - user request
+	ProcessorStatusPhaseAssigned       ProcessorStatusPhase = "assigned"       // set by kas-fleet-manager - worker
+	ProcessorStatusPhaseUpdating       ProcessorStatusPhase = "updating"       // set by kas-fleet-manager - user request
+	ProcessorStatusPhaseStopped        ProcessorStatusPhase = "stopped"        // set by kas-fleet-manager - user request
+	ProcessorStatusPhaseProvisioning   ProcessorStatusPhase = "provisioning"   // set by kas-agent
+	ProcessorStatusPhaseReady          ProcessorStatusPhase = "ready"          // set by the agent
+	ProcessorStatusPhaseFailed         ProcessorStatusPhase = "failed"         // set by the agent
+	ProcessorStatusPhaseDeprovisioning ProcessorStatusPhase = "deprovisioning" // set by kas-agent
+	ProcessorStatusPhaseDeleting       ProcessorStatusPhase = "deleting"       // set by the kas-fleet-manager - user request
+	ProcessorStatusPhaseDeleted        ProcessorStatusPhase = "deleted"        // set by the agent
+)
+
+var ValidProcessorDesiredStates = []string{
+	string(ProcessorUnassigned),
+	string(ProcessorReady),
+	string(ProcessorStopped),
+	string(ProcessorDeleted),
+}
+
+var AgentProcessorStatusPhase = []string{
+	string(ProcessorStatusPhaseProvisioning),
+	string(ProcessorStatusPhaseDeprovisioning),
+	string(ProcessorStatusPhaseStopped),
+	string(ProcessorStatusPhaseReady),
+	string(ProcessorStatusPhaseFailed),
+	string(ProcessorStatusPhaseDeleted),
+}
+
+type Processor struct {
+	db.Model
+
+	NamespaceId   *string
+	CloudProvider string
+	Region        string
+	MultiAZ       bool
+
+	Name           string
+	Owner          string
+	OrganisationId string
+	Version        int64                 `gorm:"type:bigserial;index:"`
+	Annotations    []ProcessorAnnotation `gorm:"foreignKey:ProcessorID;references:ID"`
+
+	ProcessorSpec  api.JSON `gorm:"type:jsonb"`
```

Comment:
This is pretty much a _place holder` for whatever may materialise (probably the same as is used for Connector Camel DSL definitions).